### PR TITLE
Nc feat: drag and drop image from webpage and fix: allow duplicate file upload

### PR DESCRIPTION
**Pull request by rameshmane7218**
## Change Summary

- ref: #7598 
- ref: #7601

## Change type

- [x] feat: (new feature for the user, not a new feature for build script)
- [x] fix: (bug fix for the user, not a fix to a build script)
- [ ] docs: (changes to the documentation)
- [ ] style: (formatting, missing semi colons, etc; no production code change)
- [ ] refactor: (refactoring production code, eg. renaming a variable)
- [ ] test: (adding missing tests, refactoring tests; no production code change)
- [ ] chore: (updating grunt tasks etc; no production code change)

## Test/ Verification

Provide summary of changes.

## Additional information / screenshots (optional)

Anything for maintainers to be made aware of

### CODE REVIEW

**Review by dstala — APPROVED** [review]

File: packages/nc-gui/components/cell/attachment/utils.ts [hunk 1]
```diff
@@ -249,6 +322,27 @@ export const [useProvideAttachmentCell, useAttachmentCell] = useInjectionState(
...
 
+    async function getImageDataFromUrl(imageUrl: string) {
+      try {
+        const response = await fetch(imageUrl)
+        if (response.ok) {
+          if (response.headers.get('content-type')?.startsWith('image/')) {
+            return {
+              mimetype: response.headers.get('content-type') || undefined,
+              size: +(response.headers.get('content-length') || 0) || undefined,
+            } as { minetype?: string; size?: number }
```
Comment (mertmit) [inline packages/nc-gui/components/cell/attachment/utils.ts:333]: small typo in type

File: packages/nc-gui/components/cell/attachment/utils.ts [hunk 2]
```diff
@@ -249,6 +322,27 @@ export const [useProvideAttachmentCell, useAttachmentCell] = useInjectionState(
...
+          } else if (imageUrl.slice(imageUrl.lastIndexOf('.') + 1).toLowerCase().length) {
+            return {
+              mimetype: `image/${imageUrl.slice(imageUrl.lastIndexOf('.') + 1).toLowerCase()}`,
+              size: +(response.headers.get('content-length') || 0) || undefined,
+            } as { minetype?: string; size?: number }
```
Comment (mertmit) [inline packages/nc-gui/components/cell/attachment/utils.ts:338]: small typo in type

File: packages/nocodb/src/services/public-datas.service.ts [hunk 3]
```diff
@@ -347,6 +359,65 @@ export class PublicDatasService {
...
+      );
+
+      let attachmentPath: string | undefined;
+
+      // if `attachmentUrl` is null, then it is local attachment
+      if (!attachmentUrl) {
+        // then store the attachment path only
+        // url will be constructed in `useAttachmentCell`
+        attachmentPath = `download/${filePath.join('/')}/${fileName}`;
+      }
```
Comment (mertmit) [inline packages/nocodb/src/services/public-datas.service.ts:404]: I think you refer base64 image url here but we are generating signed urls for attachments (in cloud & if NC_SECURE_ATTACHMENTS=true)
Also if you provide a filetype other than image (pdf, document etc.) it won't be available.

I see you are trying to resolve storage api being not available as public api, but we should reconsider this approach.

**Review by mertmit — DISMISSED** [review]
We can discuss further via Slack together with @dstala

**Pull request merged**